### DISCUSSION TITLE
Auto zoom / follow gps point

### DISCUSSION THREAD

**Discussion by Brandjuh** [gpx-animator/gpx-animator#480] [Ideas]
Hello,

I just recently discovered your amazing tool. Easy to use, no nonsense and fast. 

Now I thought it would be fun to combine all the GPS tracks I have into a large file and make a nice animation out of it.

The extreme points are so far apart that the map is zoomed out so far that you no longer see a lot of other tracks.

My idea is therefore to have an option that the "Camera" follows the gps point at a dynamic or defined height. At the end of the animation an overview of the complete track.

I hope my idea is a bit clear.

Thank you in advance!

**Comment by Brandjuh** [comment]
I have found a workaround for this via https://github.com/gpx-animator/gpx-animator/issues/413
However it does not work that great for me.

**Comment by McPringle** [comment]
Welcome @Brandjuh

> Now I thought it would be fun to combine all the GPS tracks I have into a large file and make a nice animation out of it.

You can add multiple GPX files to one animation, you do not need to combine them into one large file.

> The extreme points are so far apart that the map is zoomed out so far that you no longer see a lot of other tracks.
> 
> My idea is therefore to have an option that the "Camera" follows the gps point

This feature is available, it is called "moving map". If the dimensions of the map ares bigger than the dimensions of the video, the map will move. An example of a moving map can be see on on ClipTube: https://cliptube.org/w/pzyRzYFBVh5NEEJeiPaafr

You have two possibilities to create a moving map:

1. Specify the dimensions of the width and height of the map bigger than the viewport dimensions. The viewport is like a "window to the map". The zoom level of the map needed will be calculated automatically.

![image](https://user-images.githubusercontent.com/1254039/222981549-8ad6326d-7331-43e0-b238-afa5715b7f3e.png)

2. Specify the dimensions for the resulting animation and a zoom level. The size of the background map will be calculated automatically.

![image](https://user-images.githubusercontent.com/1254039/222981917-f792825d-c0ae-4b1d-895a-ef6a0924f13a.png)

> at a dynamic or defined height.

GPX Animator does not create the maps itself. The maps are provided by many providers all over the world. With "height" I think you mean the zoom level. All map providers render the map in small "tiles" in mostly 16 different zoom levels. You can specifically select the wanted zoom level. 

> At the end of the animation an overview of the complete track.

There is an open issue for this feature request: #289 

Hopefully someone will implement this at some time... :)
